Assignment 5.Rmd

```diff
@@ -40,41 +38,62 @@ We want to see if we can build a decision tree to help teachers decide which stu
 
 #Visualize our outcome variable "score"
 ```{r}
-
+hist(D1$score)
 ```
 
 #Create a categorical outcome variable based on student score to advise the teacher using an "ifelse" statement
 ```{r}
-D1$advice <- 
+D1$advice <- ifelse(D1$score<=1/3,"intervene",ifelse(D1$score<=2/3," monitor ","no action"))
+
 ```
 
 #Build a decision tree that predicts "advice" based on how many problems students have answered before, the percentage of those problems they got correct and how many hints they required
 ```{r}
-score_ctree <- 
+library(party)
+#score_ctree <- rpart(advice ~ prior_prob_count + prior_percent_correct+hints, method="class", data=D1)
+score_ctree <- ctree(factor(advice) ~ prior_prob_count + prior_percent_correct+hints, data=D1)
 ```
 
 #Plot tree
 ```{r}
-
+plot(score_ctree, title = "Session Advice Action: 1 - teacher should intervene, 2 - teacher should monitor student progress, 3 - no action")
 ```
 
-Please interpret the tree, which two behaviors do you think the teacher should most closely pay attemtion to?
+Please interpret the tree, which two behaviors do you think the teacher should most closely pay attention to?
+
+Teacher should most pay attention to these who use over 12 hints.
+
 
 #Test Tree
 Upload the data "intelligent_tutor_new.csv". This is a data set of a differnt sample of students doing the same problems in the same system. We can use the tree we built for the previous data set to try to predict the "advice" we should give the teacher about these new students. 
 
 ```{r}
 #Upload new data
 
-D2 <- 
+D2 <- read.csv("intelligent_tutor_new.csv",header = TRUE)
 
 #Generate predicted advice using the predict() command for new students based on tree generated from old students
 
-D2$prediction <- 
+D2$prediction <- predict(score_ctree,D2)
+
 
 ``` 
 ## Part III
 Compare the predicted advice with the actual advice that these students recieved. What is the difference between the observed and predicted results?
+```{r}
+length(which(D2$score==1))/length(D2$score)
+
+#predicted advice of "no action"
+length(which(D2$prediction=="no action"))
+
+#prediction accuracy rate
+length(which(D2$prediction=="no action"))/length(D2$prediction)
+
+#difference between the observed and predited results
+1-length(which(D2$prediction=="no action"))/length(D2$prediction)
+
+```
+For the new intelligent tutor data, all of students have score with value 1, which means teacher do not need to do any action for thoes students. So the number of actual advices of "no action" for these studen is 200. And the predicted advice with "no action" is 173. So the prediction accuracy rate is 86.5%. The difference between the observed and predited results is 13.5%.
```

Comment:
What does this say about the model? Does it overfit the data or is it more generalizable? Nonetheless, great job overall!